docs/catalog.md

```diff
@@ -0,0 +1,76 @@
+
+The `catalog.yaml` goal is to serve as a blueprint to find the resources managed on the respective repository, as well to provide information for software supply chain attestation, and describe continuous integration test-cases.
+
+During the release of these repositories the `catalog.yaml` is added to the payload in order to describe the Tekton resource artifacts.
+
+# Use-Cases
+
+The file described on this document is meant to make possible the use-cases described below.
+
+## Repository Root
+
+The primary location for the `catalog.yaml` file is on the root of the (Git?) repository, describing all the elements, providing software supply chain attestation data and as well descring test cases.
+
+For repositories containing the direct YAML payload of Tekton resource files stored, the file will also contain `.catalog.resources` entries, reflecting the location of the data.
+
+### Release Artifacts
+
+The `catalog.yaml` should be present on the repositories release payload, therefore when the maintainers decide to release a new version, the `catalog.yaml` is able to overwrite the entries on `.catalog.resources`.
+
+This ability makes possible to template Tekton resources instead of the plain YAML files, and during the regular releases the resource are assembled.
+
+## Continuous Integration
+
+# `catalog.{yml,yaml}`
+
+The file looks like the example below.
+
+```yml
+---
+version: v1
+
+catalog:
+  repository:
+    description: Tekton Task to interact with Git repositories
+  attestation:
+    publicKey: path/to/public.key
+    annotations:
+      team: tekton-ecosystem
+  resources:
+    tasks:
+      - name: task-git
+        version: "0.0.1"
+        filename: path/to/resource.yaml
+        sha256sum: resource-sha256-checksum
+        tri: git://openshift-pipelines/task-git@0.0.1
+        bundle: ghcr.io/openshift-pipelines/task-git:0.0.1-bundle
+        signature: path/to/signature.sig
+    pipelines: []
+```
+
+The support for the contract file is based on the `version` attribute, as this project moves forward we might change the attributes and the contract version marks breaking changes.
+
+## Repository Metadata (`.catalog.repository`)
+
+Attributes under `.catalog.repository` are meant to describe the repository containing Tekton resources, the `.description` should share a broad view of what the repository contains, what the user will find using the repository contents.
+
+## Supply Chain Attestation (`.catalog.attestation`)
+
+For the software supply chain security, the `.catalog.attestation` hols the elements needed to verify the authors signature. Initially it will contain the public key, either as a direct string or a file, and annotations for the verification processes.
+
+## Tekton Pipeline Resources (`.catalog.resources`)
```

Comment:
I noticed the presence of `.catalog.probe` in the catalog.yaml file. Shouldn't we mention it in this document, and what is its purpose?"

Reply:
I removed it recently, so what is in some catalog.yaml file might have been created with previous version of the tool.